File: 3594-IdentifyTheLargestOutlierInAnArray/3594-IdentifyTheLargestOutlierInAnArray.java
// Last updated: 1/12/2026, 3:13:15 PM
class Solution {
    public int getLargestOutlier(int[] nums) {
        int sum = 0;
        HashMap<Integer,Integer> map = new HashMap<>();
        for (int i = 0;i < nums.length;i++) {
            sum += nums[i];
            map.put(nums[i],map.getOrDefault(nums[i],0)+1);
        }
        int remainingSum;
        int max = Integer.MIN_VALUE;
        for (int i = 0;i < nums.length;i++) {
            remainingSum = sum - nums[i];
            if (remainingSum % 2 == 0) {
                remainingSum = remainingSum / 2;
                if (remainingSum == nums[i]) {
                    if (map.containsKey(remainingSum) && map.get(remainingSum) > 1) {
                        max = Math.max(nums[i],max);
                    }
                }
                else {
                    if (map.containsKey(remainingSum)) {
                        max = Math.max(nums[i],max);
                    }
                }
            }

        }
        return max;
    }
}

File: 3594-IdentifyTheLargestOutlierInAnArray/3594-IdentifyTheLargestOutlierInAnArray.py
# Last updated: 1/12/2026, 3:13:13 PM
class Solution:
    def getLargestOutlier(self, nums: List[int]) -> int:
        ret = -float('inf')
        tot = sum(nums)
        c = Counter(nums)
        for n in nums:
            if (tot-n)%2 == 0:
                if (tot-n)//2 in c:
                    if n == (tot-n)//2:
                        if c[(tot-n)//2] > 1:
                            ret = max(ret, n)
                    else:
                        ret = max(ret, n)
        
        return ret
